openapi3gen/openapi3gen.go

```diff
@@ -291,6 +321,14 @@ func (g *Generator) generateWithoutSaving(parents []*theTypeInfo, t reflect.Type
 			schema.Type = &openapi3.Types{"string"}
 			schema.Format = "date-time"
 		} else {
+			typeName := g.generateTypeName(t)
+
+			if g.opts.exportComponentSchemas.ExportComponentSchemas && g.componentSchemaRefs[typeName] != struct{}{} {
```

Comment:
Please use `ok, _ := g.componentSchemaRefs[typeName]; ok {` syntax, this allows moving the `typeName :=` within the option check.

Reply:
Have modified it, not sure what you mean by _moving the `typeName :=` within the option check._